core/controllers/editor_test.py

```diff
@@ -2937,3 +3016,127 @@ def test_rights_handler_returns_appropriate_rights(self):
         self.assertFalse(response['can_manage_voice_artist'])
 
         self.logout()
+
+
+class ImageUploadHandler(BaseEditorControllerTests):
+    """Test the handling of uploads by image uploads handler."""
+
+    def test_return_error_when_image_not_uploaded(self):
+        """Test error when no image is uploaded."""
+
+        self.login(self.EDITOR_EMAIL)
+        csrf_token = self.get_new_csrf_token()
+
+        exp_id = exp_fetchers.get_new_exploration_id()
+        self.save_new_valid_exploration(exp_id, self.editor_id)
+
+        filename = 'image_file.svg'
+        filename_prefix = 'image'
+
+        publish_url = '%s/%s/%s' % (
+            feconf.EXPLORATION_IMAGE_UPLOAD_PREFIX,
+            feconf.ENTITY_TYPE_EXPLORATION, exp_id)
+
+        response = self.post_json(
+            publish_url, {
+                'image': 'sample_image',
+                'filename': filename,
+                'filename_prefix': filename_prefix
+            },
+            csrf_token=csrf_token,
+            expected_status_int=400
+        )
+
+        error_msg = ('No image supplied')
+        self.assertEqual(response['error'], error_msg)
+
+        self.logout()
+
+    def test_return_error_when_uploaded_image_already_exists(self):
+        """Test error when uploaded image already exists."""
+
+        self.login(self.EDITOR_EMAIL)
+        csrf_token = self.get_new_csrf_token()
+
+        exp_id = exp_fetchers.get_new_exploration_id()
+        self.save_new_valid_exploration(exp_id, self.editor_id)
+
+        filename = 'img.png'
+        filename_prefix = 'image'
+
+        publish_url = '%s/%s/%s' % (
+            feconf.EXPLORATION_IMAGE_UPLOAD_PREFIX,
+            feconf.ENTITY_TYPE_EXPLORATION, exp_id)
+
+        # To get raw image for testing.
+        with python_utils.open_file(
+            os.path.join(feconf.TESTS_DATA_DIR, 'img.png'),
+            'rb', encoding=None
+        ) as f:
+            raw_image = f.read()
+
+        get_image_exits_swap = self.swap_to_always_return(
+            fs_domain.GcsFileSystem, 'isfile', value=True)
+
+        test_app_swap = self.swap(self, 'testapp', self.testapp)
```

Comment:
I don't think this does anything. You're swapping `self.testapp` into `self.testapp`

ditto below

Reply:
I removed it. here and everywhere else